File: boj/boj1000-boj1999/boj1316/cpp/00-starswot.cpp
#include <iostream>
#include <string>
using namespace std;

int isGroupWord(string s)
{
    int index[256];
    for (int i = 0; i < 256; i++)
    { index[i] = 0; }
    for (int i = 0; i < s.size(); i++)
    {
        index[s[i]]++;
        while (s[i] == s[i + 1])
        { i++; }
    }
    for (int i = 0; i < 256; i++)
    {
        if (index[i] >= 2)
        { return 0; }
    }
    return 1;
}

int main()
{
    int N;
    cin >> N;
    int answer = 0;
    for (int i = 0; i < N; i++)
    {
        string s;
        cin >> s;
        answer += isGroupWord(s);
    }
    cout << answer;
    return 0;
}

File: srcs/15650/js/00-starswot.js
const [N, M] = require('fs')
                .readFileSync('../input/1.txt')
                .toString()
                .trim()
                .split(' ')
                .map(Number);

let answer = '';
let stack = [];

function solution(index, cnt) {
    if (cnt == M) {
        answer += stack.join(' ') + '\n';
        return ;
    }
    for (let i = index; i <= N; i++) {
        stack.push(i);
        solution(i + 1, cnt + 1);
        stack.pop();
    }
}

solution(1, 0);
console.log(answer);
